code/modules/mob/living/carbon/human/life.dm

```diff
@@ -318,4 +320,88 @@
 		adjustOxyLoss(5)
 		adjustBruteLoss(1)
 
+/*
+Alcohol Poisoning Chart
+Note that all higher effects of alcohol poisoning will inherit effects for smaller amounts (i.e. light poisoning inherts from slight poisoning)
+In addition, severe effects won't always trigger unless the drink is poisonously strong
+All effects don't start immediately, but rather get worse over time; the rate is affected by the imbiber's alcohol tolerance
+
+0: Non-alcoholic
+1-10: Barely classifiable as alcohol - occassional slurring
+11-20: Slight alcohol content - slurring
+21-30: Below average - imbiber begins to look slightly drunk
+31-40: Just below average - no unique effects
+41-50: Average - mild disorientation, imbiber begins to look drunk
+51-60: Just above average - disorientation, vomiting, imbiber begins to look heavily drunk
+61-70: Above average - small chance of blurry vision, imbiber begins to look smashed
+71-80: High alcohol content - blurry vision, imbiber completely shitfaced
+81-90: Extremely high alcohol content - heavy toxin damage, passing out
+91-100: Dangerously toxic - swift death
+*/
+
+/mob/living/carbon/human/proc/handle_alcohol()
+	if(sleeping)
+		modify_drunkenness(-1.5) //Sleeping helps you recover from alcohol poisoning
+	else
+		modify_drunkenness(-0.2) //Becoming sober takes much longer than becoming drunk
+
+	if(drunkenness >= 1)
+		if(prob(25))
+			slurring += 2
+		jitteriness = max(jitteriness - 3, 0)
+
+	if(drunkenness >= 11)
+		slurring += 1.2 //Constant but only a very slow increase
+
+	if(drunkenness >= 21)
+		drunk_state = LIGHTLY_DRUNK
+
+	if(drunkenness >= 41)
+		if(prob(25))
+			confused += 2
+		Dizzy(10)
+		drunk_state = DRUNK
+
+	if(drunkenness >= 51)
+		if(prob(5))
+			vomit()
+		Dizzy(25)
+		confused += 1.2
+		drunk_state = HEAVILY_DRUNK
+
+	if(drunkenness >= 61)
+		if(prob(50))
+			adjust_blurriness(5)
+		drunk_state = SMASHED
+
+	if(drunkenness >= 71)
+		adjust_blurriness(1.2)
+		drunk_state = SHITFACED
+
+	if(drunkenness >= 81)
+		adjustToxLoss(0.2)
+		if(prob(5))
+			src << "<span class='warning'>Maybe you should lie down for a bit...</span>"
+
+	if(drunkenness >= 91)
+		if(!sleeping)
+			adjustToxLoss(1)
+		drunk_state = BARELY_ALIVE
+		if(prob(20))
+			if(SSshuttle.emergency.mode == SHUTTLE_DOCKED && z == ZLEVEL_STATION) //QoL mainly
+				src << "<span class='warning'>You're so tired... but you can't miss that shuttle...</span>"

```

Comment:
I knew that people would whine about passing out because they were about to board the shuttle and get greentext and then passed out from RNG. I could remove it if you want me to.